magefile.go

```diff
@@ -140,7 +142,7 @@ func TestUnit() {
 }
 
 func Build() {
-	rebuild, err := target.Path(filepath.Join(binDir, "kubernetes"), srcDirs...)
+	rebuild, err := target.Dir(filepath.Join(binDir, "kubernetes"), srcDirs...)
```

Comment:
I think the `SrcDir` contains directory as well. Based on the documentation of `target.Dir`,
 `// Dir reports whether any of the sources have been modified more recently
// than the destination. If a source or destination is a directory, this
// function returns true if a source has any file that has been modified more
// recently than the most recently modified file in dst. 
`
This function will walk through all the files in the src dir instead of just rely on the modification time on the folder itself

Reply:
My point is that the full path you are calling Dir on is bin/plugins/kubernetes/kubernetes which is a file, not a directory. We are checking the file and not the directory because otherwise it skips building entirely when the kubernetes file does not exist.

Let me know if you were running into a specific problem here that prompted changing this from File to Dir. This is the only place were we are using the target package and I don't yet have a good handle on how best to use it.

Right now it's a bit clunky IMO...

Reply:
You are correct. The destination path is a file path which `target.Dir` also accepts. 
From my understanding, what we are trying to do here is to detect if any file has been modified in the source folders more recent than the destination file.
The `target.Path` only checks for the stats on the source folder itself rather than traverse all the files contained in source folder, which in my experience, only changes when the folder itself is modified.
The `target.Dir` does exactly that, traversing all the files inside the source dir to check if any files has changed since the destination file modified timestamp

Reply:
Oh! I had it mixed up in my head and thought it worked differently, thanks for clarifying 🙇‍♀️